core/admin/mailu/internal/nginx.py

```diff
@@ -50,8 +50,12 @@ def check_credentials(user, password, ip, protocol=None, auth_port=None, source_
                     app.logger.info(f'Login attempt for: {user}/{protocol}/{auth_port} from: {ip}/{source_port}: failed: badip: token-{token.id}: {token.comment or ""!r}')
                     return False # we can return directly here since the token is valid
     if user.check_password(password):
-        app.logger.info(f'Login attempt for: {user}/{protocol}/{auth_port} from: {ip}/{source_port}: success: password')
-        return True
+        if app.config['AUTH_REQUIRE_TOKENS'] and protocol != 'web':
+            app.logger.info(f'Login attempt for: {user}/{protocol}/{auth_port} from: {ip}/{source_port}: failed: password but AUTH_REQUIRE_TOKENS=True')
```

Comment:
Maybe the message should be `failed: password okay, but AUTH_REQUIRE_TOKENS=True` or `failed: password good, but token required`

Reply:
That's [2494a34](https://github.com/Mailu/Mailu/pull/3007/commits/2494a344a7101ae36cd36d4837f2caa2dcf14139)